lib/testing/matrix.js

```diff
@@ -0,0 +1,210 @@
+'use strict'; // eslint-disable-line strict
+
+/**
+* Running tests with multiple configurations using varying parameters
+* @constructor
+*/
+class TestMatrix {
+    /**
+    * Builds a new testing Matrix
+    *
+    * @param {Object.<string, string|string[]>} params - Argument of the matrix
+    *
+    * @returns {TestMatrix} new testing Matrix
+    */
+    constructor(params) {
+        /**
+        * Create deep copy for prevent the user to modify the object outside
+        * the class
+        **/
+        this.params = Object.assign({}, params);
+        this.callback = null;
+        this.elementsToSpecialize = [];
+        this.description = '';
+        this.listOfSpecialCase = [];
+    }
+
+    /**
+    * Callback used for prevent mocha that the test is over
+    * Must be call in the end of every test
+    * @callback Done
+    *
+    * @return {undefined}
+    */
+    /**
+    * Callback used for a generated matrix
+    * Used by @generate and @if method
+    *
+    * @callback GenerateMatrixCallback
+    * @param {TestMatrix} matrix - The following generate matrix.
+    * @param {Done} done - Callback for prevent mocha that we finish
```

Comment:
I don't understand this comment. Is this the `done` callback used by the test to signal mocha the end of the test?